automol/util/vector.py

```diff
@@ -270,160 +257,155 @@ def _local_axes(xyz1=(0.0, 0.0, 0.0), xyz2=(0.0, 0.0, 1.0), xyz3=(0.0, 1.0, 0.0)
+) -> float:
     """Measure the angle inscribed by three atoms,
     projected onto the normal plane of the central atom.
 
     :param xyz1: 3D vector to point 1
-    :type xyz1: tuple, list, or numpy nd.array
     :param xyz2: 3D vector to point 2
-    :type xyz2: tuple, list, or numpy nd.array
     :param xyz3: 3D vector to point 3
-    :type xyz3: tuple, list, or numpy nd.array
-    :rtype: float
+    :return: Angle projected on normal plane
     """
-
     uxyz21 = unit_perpendicular(xyz2, xyz1)
     uxyz23 = unit_perpendicular(xyz2, xyz3)
-    cos = numpy.dot(uxyz21, uxyz23)
+    cos = np.dot(uxyz21, uxyz23)
     if cos < -1.0:
-        assert numpy.allclose(cos, -1.0)
+        assert np.allclose(cos, -1.0)
         cos = -1.0
     elif cos > 1.0:
-        assert numpy.allclose(cos, 1.0)
+        assert np.allclose(cos, 1.0)
         cos = 1.0
-    ang = numpy.arccos(cos)
+    ang = np.arccos(cos)
     return ang
 
 
-def dihedral_angle(xyz1, xyz2, xyz3, xyz4):
+def dihedral_angle(
+    xyz1: Sequence | np.ndarray,
+    xyz2: Sequence | np.ndarray,
+    xyz3: Sequence | np.ndarray,
+    xyz4: Sequence | np.ndarray,
+) -> float:
     """Measure the dihedral angle defined by four atoms.
 
     :param xyz1: 3D vector to point 1
-    :type xyz1: tuple, list, or numpy nd.array
     :param xyz2: 3D vector to point 2
-    :type xyz2: tuple, list, or numpy nd.array
     :param xyz3: 3D vector to point 3
-    :type xyz3: tuple, list, or numpy nd.array
     :param xyz4: 3D vector to point 4
-    :type xyz4: tuple, list, or numpy nd.array
-    :rtype: float
+    :return: Dihedral angle of four atoms
     """
-
     # Get the cosine of the angle
     uxyz21 = unit_direction(xyz2, xyz1)
     uxyz23 = unit_direction(xyz2, xyz3)
     uxyz32 = unit_direction(xyz3, xyz2)
     uxyz34 = unit_direction(xyz3, xyz4)
     uxyz123_perp = unit_perpendicular(uxyz21, uxyz23)
     uxyz234_perp = unit_perpendicular(uxyz32, uxyz34)
-    cos = numpy.dot(uxyz123_perp, uxyz234_perp)
+    cos = np.dot(uxyz123_perp, uxyz234_perp)
 
     # Get the sign of the angle
-    val = numpy.dot(uxyz123_perp, uxyz34)
+    val = np.dot(uxyz123_perp, uxyz34)
     val = max(min(val, 1.0), -1.0)
     sign = 2 * (val < 0) - 1
 
     # Before plugging it into the arccos function, make sure we haven't
     # Slightly run out of bounds
     if cos < -1.0:
-        assert numpy.allclose(cos, -1.0)
+        assert np.allclose(cos, -1.0)
         cos = -1.0
     elif cos > 1.0:
-        assert numpy.allclose(cos, 1.0)
+        assert np.allclose(cos, 1.0)
         cos = 1.0
 
-    dih = sign * numpy.arccos(cos)
-    dih = numpy.mod(dih, 2 * numpy.pi)
+    dih = sign * np.arccos(cos)
+    dih = np.mod(dih, 2 * np.pi)
     return dih
 
 
 # transformations
 def rotator(
-    axis: Vector, ang: float, degree: bool = False, orig_xyz: Optional[Vector] = None
+    axis: Vector, ang: float, degree: bool = False, orig_xyz: Vector | None = None
 ) -> Callable[[Vector], Vector]:
```

Comment:
Looks good. To be consistent, let's do:
```
 axis: Sequence | Vector, ang: float, degree: bool = False, orig_xyz: Sequence | Vector = (0., 0., 0.)
```
and change the handling of `orig_xyz` as described below.

Also, the return value should be:
```
Callable[[Sequence | Vector], Vector]
```

Reply:
Also, we'll need to `require_vector_like` for the vector inputs.